clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java

```diff
@@ -214,6 +321,10 @@ public Errors error() {
         return responseData;
     }
 
+    public Map<TopicPartition, PartitionData> responseData(String groupId) {
```

Comment:
We should probably have a `responseData` method that takes version or ensure we have the data cached regardless of version

Reply:
I am planning on changing this method to check and see if `groupToPartitionData` is empty. If it is empty we are using a version less than the one that supports batching and it will fall back to the old behavior. If not we know the response is initialized with data that supports batching and then it returns from the information from `groupToPartitionData`

Reply:
Yes, sounds good.